components/candle/challengeService.ts

```diff
@@ -551,22 +553,30 @@ export class ChallengeService extends ChallengeRegistry {
             true,
         )
         const parent = locations.children[child].Properties.ParentLocation
-        const location = locations.children[child]
-        assert.ok(location)
-
-        let contracts = isSniperLocation(child)
-            ? this.controller.missionsInLocations.sniper[child]
-            : this.controller.missionsInLocations[child]
-        if (!contracts) {
-            contracts = []
-        }
 
-        return this.getGroupedChallengeLists(
-            {
+        let challengeFilterOptions: ChallengeFilterOptions
+
+        if (filterType === ChallengeFilterType.None) {
+            challengeFilterOptions = {
+                type: ChallengeFilterType.None,
+            }
```

Comment:
Why is `None` needed here?

Reply:
My initial frame of thought before I used the function way my own "getByLocation" function and I did not apply any other logic there other than locationId/parentLocationId check. Your filter also mentions it's only meant to be used for Career > Challenges. As such, not knowing exactly what it did, I felt it was best to not filter at all.

Reply:
https://github.com/thepeacockproject/Peacock/blob/673bfec718a1552e61474259fb16025b223386b8/components/candle/challengeHelpers.ts#L74-L75

Reply:
I think you can give that filter a try. That comment was meant to record where it is currently used, not where it can only be used.